cabby/data/wikipedia/query.py

```diff
@@ -106,129 +125,163 @@ def clean_text(text: Text) -> Text:
   return text
 
 
-def get_wikipedia_items_title_in_text(backlink_id: int, orig_title: Text) -> Sequence:
-  '''Query the Wikipedia API. 
+def get_backlinks_ids_from_wikipedia_title(title: str) -> Sequence[int]:
+  '''Query the Wikipedia API for backlinks pageids. 
   Arguments:
-    backlinks_titles(Sequence[Text]): The Wikipedia backlinks titles to
-    run on the Wikipedia API.
-    orig_title(Text): The Wikipedia title that would be searched in
-    the backlinks text.
+    title: The Wikipedia title to obtain backlinks from.
   Returns:
-    The backlinks Wikipedia items found.
+    The backlinks pageids.
   '''
-
   url = (
     'https://en.wikipedia.org/w/api.php'
     '?action=query'
-    '&prop=extracts'
-    '&explaintext'
-    f'&pageids={backlink_id}'
+    '&prop=linkshere'
+    f'&titles={title}'
     '&format=json'
+    '&lhlimit=500'
```

Comment:
I'm happy starting with 500 results from the API because I filter based on the BACKLINK_COUNT_THRESHOLD, then randomly shuffle the remainder and then take MAX_BACKLINKS_PER_TITLE from the resulting list.